src/nodes.js

```diff
@@ -143,10 +154,44 @@ var registerChild = function(parent, key, child) {
 };
 
 
+/**
+ * Marks the key map for a given Element as invalid. It will be cleaned the
+ * next time it is needed.
+ * @param {!Element} el
+ */
+var invalidateKeyMap = function(el) {
+  getKeyMap(el).isValid = false;
+};
+
+
+/**
+ * Cleans a key map for a given Element, removing any nodes that are no longer
+ * in the DOM.
+ * @param {!Element} el
+ */
+var cleanKeyMap = function(el) {
+  var keyMap = getData(el).keyMap;
+
+  if (!keyMap || keyMap.isValid) {
+    return;
+  }
+
+  for (var key in keyMap) {
+    if (!keyMap[key].parentNode) {
+      delete keyMap[key];
```

Comment:
You're right, I was using the index of a loop as the key (`'' + i`). I've updated #48 to `delete`.